### PR TITLE
Vertical Layout or Multiple Buttons (from bobwhaler's PR)

### PR DESCRIPTION
**Pull request by ryang1428**
- removed topMargin property for another PR
- Also, we still need to handle many buttons by placing the buttons in a scrollView, or one of the other options we talked about @jgrandelli. or, we can open up an issue for that
  original PR: 
  https://github.com/urbn/URBNAlert/pull/49

### CODE REVIEW

**Comment by ryang1428** [comment]
hold on this actually I have to get the example back in

**Comment by ryang1428** [comment]
i also want to test this branch out in FP

**Comment by ryang1428** [comment]
@jgrandelli Gift Message alert looked good in checkout.

**Comment by jgrandelli** [comment]
@bobawhaler do you wanna take a look at this one and see how it works for your needs? This adds the >2 button layout. The top margin stuff needs to be addressed separately. There were other layout implications.

**Comment by bobawhaler** [comment]
Ok. No problem. If you doubt in this way adding topMargin you can skip it for now.

**Comment by ryang1428** [comment]
@jgrandelli we good here then?

File: Pod/Classes/URBNAlertView.m [hunk 1]
```diff
@@ -82,11 +82,19 @@ - (instancetype)initWithAlertConfig:(URBNAlertConfig *)config alertStyler:(URBNA
         
         // Add some buttons
         NSMutableArray *btns = [NSMutableArray new];
+        NSMutableArray *separators = [NSMutableArray new];
         buttonContainer.translatesAutoresizingMaskIntoConstraints = NO;
         
         __weak typeof(self) weakSelf = self;
         [self.alertConfig.actions enumerateObjectsUsingBlock:^(URBNAlertAction *action, NSUInteger idx, BOOL *stop) {
             if (action.isButton) {
+                if ([self.alertStyler.separatorHeight boolValue]) {
```
Comment (jgrandelli) [inline Pod/Classes/URBNAlertView.m]: Not a fan of this. Converting numbers to bools doesn't feel right. Would be better to check it numerically

File: Pod/Classes/URBNAlertView.m [hunk 2]
```diff
@@ -151,18 +161,43 @@ - (instancetype)initWithAlertConfig:(URBNAlertConfig *)config alertStyler:(URBNA
...
         self.buttons = [btns copy];
+        self.sectionCount++;
         if (self.buttons.count == 1) {
-            self.sectionCount++;
             [buttonContainer addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-btnLeftMargin-[btnOne]-btnRightMargin-|" options:0 metrics:metrics views:@{@"btnOne" : self.buttons.firstObject}]];
             [buttonContainer addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-btnTopMargin-[btnOne(btnH)]-btnBottomMargin-|" options:0 metrics:metrics views:@{@"btnOne" : self.buttons.firstObject}]];
         }
-        else if (self.buttons.count == 2) {
-            self.sectionCount++;
+        else if (self.buttons.count == 2 && !self.alertStyler.useVerticalLayoutForTwoButtons.boolValue) {
```
Comment (jgrandelli) [inline Pod/Classes/URBNAlertView.m:169]: Unless I'm missing something, are you not using the button separators for 2 button vertical layouts?

File: Pod/Classes/URBNAlertView.m [hunk 3]
```diff
@@ -189,7 +233,7 @@ - (void)layoutSubviews {
         screenHeight = screenSize.width;
     }
     
-    CGFloat maxHeight = screenHeight - self.titleLabel.intrinsicContentSize.height - (self.alertStyler.sectionVerticalMargin.floatValue * self.sectionCount) - buttonHeight - kURBNAlertViewHeightPadding;
+    CGFloat maxHeight = screenHeight - self.titleLabel.intrinsicContentSize.height - (self.alertStyler.sectionVerticalMargin.floatValue * self.sectionCount) - buttonHeight * numberOfVerticalButtons - kURBNAlertViewHeightPadding;
```
Comment (jgrandelli) [inline Pod/Classes/URBNAlertView.m]: Are you taking into account here any button separators?

File: Pod/Classes/URBNAlertView.m (hunk 2 above)
Comment (ryang1428) [inline Pod/Classes/URBNAlertView.m:169]: after looking at this it does-- if there are 2 buttons and `self.alertStyler.useVerticalLayoutForTwoButtons` is true, it goes to the next block which adds the separators

**Comment by ryang1428** [comment]
- Re-tested in FP, including the checkout alerts-- all seemed ok. I did not see any spacing issues

**Pull request merged**